### DISCUSSION TITLE
User to Manager self relation

### DISCUSSION THREAD

**Discussion by SleeveShirtholes** [prisma/prisma#19581] [Q&A] [answered]
I'm trying to create a user table with the following structure but I can't get the self relation to work and it stops me from being able to create a migration and the tables in the database. I'm using the example here https://www.prisma.io/docs/concepts/components/prisma-schema/relations/self-relations as a guide.

If I uncomment the manager relation, I get the following error:

`Error validating field `manager` in model `User`: The relation field `manager` on model `User` is missing an opposite relation field on the model `User`. Either run `prisma format` or add it manually.`

```prisma
model User {
  id           Int           @id @default(autoincrement())
  name         String?
  email        String        @unique
  managerId    Int?
  jobTitleId   Int
  departmentId Int
  createdAt    DateTime      @default(now())
  updatedAt    DateTime      @updatedAt
  jobTitle     JobTitle      @relation(fields: [jobTitleId], references: [id])
  department   Department    @relation(fields: [departmentId], references: [id])
  userHistory  UserHistory[]
  // manager      User?         @relation("Manager", fields: [managerId], references: [id])
}
```

Please help!

PS. If I use the prisma auto formatting in vscode it adds a bunch of fields that I believe are unnecessary.

**Comment by SleeveShirtholes** [comment]
I forgot to mention that I also tried doing what the error says like so:

```
model User {
  id           Int           @id @default(autoincrement())
  name         String?
  email        String        @unique
  managerId    Int?
  jobTitleId   Int
  departmentId Int
  createdAt    DateTime      @default(now())
  updatedAt    DateTime      @updatedAt
  jobTitle     JobTitle      @relation(fields: [jobTitleId], references: [id])
  department   Department    @relation(fields: [departmentId], references: [id])
  userHistory  UserHistory[]
  manager      User?         @relation("Manager", fields: [managerId], references: [id])
  managers User[]
}
```

But I receive this error:

`Error validating field `managers` in model `User`: The relation field `managers` on model `User` is missing an opposite relation field on the model `User`. Either run `prisma format` or add it manually.`

**Reply by ludralph** [reply, marked as answer]
Hi @SleeveShirtholes 👋 

It seems you're facing an issue with setting up the self-relation in your User model using Prisma. The error message suggests that you need to add an opposite relation field on the model User to establish the self-relation. Here's how you can modify your code to resolve the issue:
```prisma
model User {
  id           Int           @id @default(autoincrement())
  name         String?
  email        String        @unique
  managerId    Int?
  jobTitleId   Int
  departmentId Int
  createdAt    DateTime      @default(now())
  updatedAt    DateTime      @updatedAt
  jobTitle     JobTitle      @relation(fields: [jobTitleId], references: [id])
  department   Department    @relation(fields: [departmentId], references: [id])
  userHistory  UserHistory[]
  manager      User?         @relation("Manager", fields: [managerId], references: [id])
  employees    User[]        @relation("Manager")
}
```
In the modified code, I've added the manager field back to the User model and created an opposite relation field called employees. This establishes a self-relation where a user can have many employees (as a manager), and each employee belongs to a single manager.

**Reply by SleeveShirtholes** [reply]
Awesome! That worked! Thank you for the help @ludralph!
